src/components/Attributes/utils.ts

```diff
@@ -76,6 +76,17 @@ export function getReferenceDisplayValue(
       return definedAttributeReference;
     }
 
+    // If value has not been yet assigned and data
+    // is no longer available, use metadata
+    if (attribute.metadata) {
+      return {
+        label: attribute.metadata.find(
+          metadata => metadata.value === attributeValue,
+        )?.label,
```

Comment:
question: why you keep metadata in array instead of an object?

Reply:
We could choose `key:value` as `id:label` pairs here but it's simpler to translate `Container[]` type returned from Assign Dialogs to array than single object